docs/src/arrays.md

```diff
@@ -0,0 +1,440 @@
+
+```julia
+function rrule(::typeof(*), A::Matrix{<:RealOrComplex}, B::Matrix{<:RealOrComplex})
+    function times_pullback(ΔC)
+        ∂A = @thunk(ΔC * B')
+        ∂B = @thunk(A' * ΔC)
+        return (NO_FIELDS, ∂A, ∂B)
+    end
+    return A * B, times_pullback
+end
+```
+
+### Matrix inversion
+
+```julia
+C = inv(A)
+```
+
+In \eqref{invdiff}, we derived the differential identity as
+
+$$dC = -C ~dA~ C$$
+
+Multiplying by $\overline{C}^H$ and taking the real trace,
+
+```math
+    ds = \operatorname{real}\left( \operatorname{tr}(\overline{C}^H ~dC) \right)
+       = \operatorname{real}\left( \operatorname{tr}(-\overline{C}^H C ~dA~ C) \right)
+```
+
+Applying the trace identity \eqref{trperm} to manipulate into the desired form,
+
+$$ds = \operatorname{real}(\operatorname{tr}(-C \overline{C}^H C ~dA) ),$$
+
+we can now solve for $\overline{A}$:
+
+$$\overline{A} = (-C \overline{C}^H C)^H = -C^H \overline{C} C^H$$
+
+We can implement the resulting `rrule` as:
+
+```julia
+function rrule(::typeof(inv), A::Matrix{<:RealOrComplex})
+    C = inv(A)
+    function inv_pullback(ΔC)
+        ∂A = -C' * ΔC * C'
+        return (NO_FIELDS, ∂A)
+    end
+    return C, inv_pullback
+end
+```
+
+## More examples
+
+For more instructive examples of matrix and vector rules, see [^Giles2008ext] and the [LinearAlgebra rules in ChainRules](https://github.com/JuliaDiff/ChainRules.jl/tree/master/src/rulesets/LinearAlgebra).
+
+## Generalizing to multidimensional arrays
+
+For both forward- and reverse-mode rules for matrices, the first step was to write down the differential identities, which followed directly from the scalar differential identities.
+This approach follows for arrays, but it's easier to work in component form.
+Consider the following function
+
+```julia
+C = sum(abs2, A::Array{<:RealOrComplex,3}; dims=2)::Array{<:Real,3}
+```
+
+which we write as
+
+$$C_{i1k} = \sum_{j} |A_{ijk}|^2 = \sum_{j} \operatorname{real}(\operatorname{conj}(A_{ijk}) A_{ijk})$$
+
+The differential identity is
+
+```math
+\begin{align*}
+    dC_{i1k} &= \sum_j \operatorname{real}( \operatorname{conj}(dA_{ijk})~ A_{ijk} +
+                            \operatorname{conj}(A_{ijk}) ~dA_{ijk} ) \\
+             &= \sum_j \operatorname{real}(
+                    \operatorname{conj}\left(
+                        \operatorname{conj}(A_{ijk}) ~dA_{ijk}
+                    \right) +
+                        \operatorname{conj}(A_{ijk}) ~dA_{ijk}
+                )\\
+             &= \sum_j 2 \operatorname{real}( \operatorname{conj}(A_{ijk}) ~dA_{ijk} )
+\end{align*}
+```
+
+where in the last step we have used the fact that for all real $a$ and $b$,
+
+$$(a + i b) + \operatorname{conj}(a + i b) = (a + i b) + (a - i b) = 2 a = 2 \operatorname{real} (a + i b).$$
+
+The `frule` can be implemented generally as
+
+```julia
+function frule(
+    (_, _, ΔA),
+    ::typeof(sum), ::typeof(abs2), A::Array{<:RealOrComplex};
+    dims = :,
```

Comment:
should `dims=2`? since the primal function [shown above](https://github.com/JuliaDiff/ChainRulesCore.jl/pull/191/files#diff-04495d68483dba4eac9bf4246b9c62d0R342) has `dims=2`?

```suggestion
    dims=2,
```